### PR TITLE
Update DataStoreImpl.java

### PR DESCRIPTION
**Pull request by loganullmeyer**

### CODE REVIEW

**Comment by loganullmeyer** [comment]
This checkstyle isn't passing over one of the sample solution methods

**Comment by Oliver-222** [comment]
> This checkstyle isn't passing over one of the sample solution methods

This seems to be the error that checkstyle is getting:
error line="88" column="33" severity="warning" message="Avoid using finalizer method." source="com.puppycrawl.tools.checkstyle.checks.coding.NoFinalizerCheck"

This is in: FactorAPI/DataStoreImpl

Once that is edited I think it should be good.

File: src/FactorAPI/DataStoreImpl.java [hunk 1]
```diff
@@ -1,69 +1,141 @@
...
+				
+				/*
+				 * finalize() is a method on Object, much like toString or equals. It's called when an object is garbage collected, and is used as
+				 * a final cleanup step for resources. It's a bit fragile - it isn't guaranteed to always be called, or be called at any specific time - 
+				 * but unless a cleanup is particularly critical, it's generally sufficient as a back-stop against weird circumstances. In this case,
+				 * we would at worst leak a read-lock file handle, which is NBD and certainly not worth architecting a larger solution around (honestly,
+				 * finalize() might even be overkill in this situation).
+				 */
+				@Override
+				public void finalize() {
```
Comment (Oliver-222) [inline src/FactorAPI/DataStoreImpl.java]: Checkstyle doesn't want the finalize method used it seems.

**Review by Oliver-222 — APPROVED** [review]
Overall looks good just look into that finalize line as the checkstyle doesn't like it. Only merge this once all tests are passing. Other files look good.

**Comment by loganullmeyer** [comment]
It seems like the finalize() method isn't super necessary, so for now I removed it, and I'll ask about it later on

**Pull request merged**